packages/dynamics-lib/src/entities/base.entity.js

```diff
@@ -163,8 +169,13 @@ export class BaseEntity {
    * @private
    */
   _setOptionSet (property, value) {
-    const mapping = this.constructor.definition.mappings[property]
-    if (!(value instanceof GlobalOptionSetDefinition) || value.optionSetName !== mapping.ref) {
+    if (
+      !(
+        Object.prototype.hasOwnProperty.call(value, 'id') &&
+        Object.prototype.hasOwnProperty.call(value, 'label') &&
+        Object.prototype.hasOwnProperty.call(value, 'description')
```

Comment:
Could you not just do `value.hasOwnProperty('id') && value.hasOwnProperty('label') && value.hasOwnProperty('description')`?

Also, 
```
const validGlobalOptionSetDefinition = !(value.hasOwnProperty('id') && value.hasOwnProperty('label') && value.hasOwnProperty('description'))
if (validGlobalOptionSetDefinition) { //...
```
might be a bit clearer and more expressive

Reply:
Eslint warns about doing `value.hasOwnProperty('id') && value.hasOwnProperty('label') && value.hasOwnProperty('description')`

Reply:
Fair enough - have a think about my second comment on `const validGlobalOptionSetDefinition`